src/App.tsx

```diff
@@ -8,52 +8,50 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
-import { getUserPosts } from './api/posts';
-import { User } from './types/User';
-import { Post } from './types/Post';
-import { Counter } from './features/counter/Counter';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import * as postActions from './features/postsSlice';
+import * as userActions from './features/usersSlice';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
-
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const dispatch = useAppDispatch();
+  const {
+    items: posts,
+    loaded,
+    hasError,
+    selectedPost,
+  } = useAppSelector(state => state.posts);
+  const { users, author } = useAppSelector(state => state.users);
```

Comment:
It's better to create a separate selectors so you can reuse them
https://redux.js.org/usage/deriving-data-selectors